cuda_core/tests/test_program.py

```diff
@@ -37,9 +74,7 @@ def test_program_init_invalid_code_format():
 def test_program_compile_valid_target_type():
```

Comment:
Could you add a comment in this test noting that we're testing the capability of using the current device arch?